sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -133,8 +142,77 @@ object ScalaJSPluginInternal {
       }
   )
 
+  private def scalajspSettings: Seq[Setting[_]] = {
+    case class Options(
+        infos: Boolean = false,
+        showReflProxy: Boolean = false
+    )
+
+    val optionsParser: Parser[Options] = {
+      token(OptSpace ~> (
+          (literal("-i") | "--infos") ^^^ ((_: Options).copy(infos = true))
+        | (literal("-p") | "--reflProxies") ^^^ ((_: Options).copy(showReflProxy = true))
+      )).* map {
+        fns => fns.foldLeft(Options())((prev, fn) => fn(prev))
+      }
+    }
+
+    def sjsirFileOnClasspathParser(
+        relPaths: List[String]): Parser[String] = {
+      OptSpace ~> StringBasic
+        .examples(ScalajspUtils.relPathsExamples(relPaths))
+    }
+
+    def scalajspParser(state: State, relPaths: List[String]) =
+      optionsParser ~ sjsirFileOnClasspathParser(relPaths)
+
+    val parser = loadForParser(sjsirFilesOnClasspath) { (state, relPaths) =>
+      scalajspParser(state, relPaths.getOrElse(Nil))
+    }
+
+    Seq(
+        sjsirFilesOnClasspath <<= Def.task {
+          val cp = Attributed.data(fullClasspath.value)
+          ScalajspUtils.listSjsirFilesOnClasspath(cp)
+        } storeAs(sjsirFilesOnClasspath) triggeredBy(fullClasspath),
+
+        scalajsp := {
+          val (options, relPath) = parser.parsed
+
+          val cp = Attributed.data(fullClasspath.value)
+          val vfile = ScalajspUtils.loadIRFile(cp, relPath)
+
+          val stdout = new java.io.PrintWriter(System.out)
+          if (options.infos) {
+            new InfoPrinter(stdout).printClassInfo(vfile.info)
+          } else {
+            val (info, tree) = vfile.infoAndTree
+            val outTree = {
+              if (options.showReflProxy) tree
+              else filterOutReflProxies(tree)
+            }
+            new IRTreePrinter(stdout).printTopLevelTree(outTree)
+          }
+          stdout.flush()
+        }
+    )
+  }
+
+  // !!! CODE DUPLICATION with Scalajsp.filterOutReflProxies
+  private def filterOutReflProxies(tree: ir.Trees.ClassDef): ir.Trees.ClassDef = {
+    import ir.Trees._
+    import ir.Definitions.isReflProxyName
+    val newDefs = tree.defs.filter {
+      case MethodDef(_, Ident(name, _), _, _, _) => !isReflProxyName(name)
+      case _ => true
+    }
+    tree.copy(defs = newDefs)(tree.optimizerHints)(tree.pos)
+  }
+
```

Comment:
I did consider it, but I don't think it's generally useful. In theory one should never remove those. It's only useful for display. Besides, if they're in tools or ir, it becomes a public API, and I don't think this should be a public API, for the reasons outlined in the first sentence.